### PR TITLE
fix: add logic for creating schedules for recurring donations

### PR DESCRIPTION
**Pull request by linnall**
## Notion ticket link

[Editing the create schedule endpoint](https://www.notion.so/uwblueprintexecs/Editing-the-create-schedule-endpoint-847a758ef1b04f8db226e0a79a2519fa)

## Implementation description
* add logic for creating schedules for recurring donations

## Steps to test
1. use Postman to create schedules with `frequency` daily, weekly, and monthly
2. verify in pgadmin if the correct number of objects were created with the correct fields

## What should reviewers focus on?
* correctness of values in database

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/services/implementations/schedulingService.ts [hunk 1]
```diff
@@ -133,23 +147,183 @@ class SchedulingService implements ISchedulingService {
...
+        });
+
+        const arrayRecurringDonationIds: number[] = [];
+        for (let i = 0; i < recurringDonationIds.length; i += 1) {
+          if (recurringDonationIds[i] === "null") {
+            arrayRecurringDonationIds.push(0);
+          } else {
+            arrayRecurringDonationIds.push(Number(recurringDonationIds[i]));
+          }
+        }
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:183]:
```suggestion
       recurringDonationIds.forEach((recurringDonationId, i)=> {
       arrayRecurringDonationIds.push(recurringDonationIds[i] ? Number(recurringDonationid) : 0)
       })
```

File: backend/typescript/services/implementations/schedulingService.ts [hunk 2]
```diff
@@ -133,23 +147,183 @@ class SchedulingService implements ISchedulingService {
...
+
+        const newRecurringDonationId: number = arrayRecurringDonationIds.length
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:185]:
```suggestion
        const newRecurringDonationId: number = arrayRecurringDonationIds.length > 0
```

**Review by hanlinc27 — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/WeeklyCalendar.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import { HStack, Text, VStack } from "@chakra-ui/react";
```
Comment (Shehryar21) [inline frontend/src/components/common/WeeklyCalendar.tsx:1]: we can delete this file? Akshay already moved to another folder.

File: frontend/src/components/common/WeeklyEventItemPopUp.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
+import {
```
Comment (Shehryar21) [inline frontend/src/components/common/WeeklyEventItemPopUp.tsx:1]: Same with this. We can delete this.

**Review by Shehryar21 — COMMENTED** [review]
I still need to review. Just comments to delete before.

**Review by Shehryar21 — APPROVED** [review]

**Review by Shehryar21 — APPROVED** [review]

File: backend/typescript/services/implementations/schedulingService.ts (hunk 1 above)
Comment (linnall) [reply backend/typescript/services/implementations/schedulingService.ts:184]: Will refactor all logic for recurring schedule creation post-launch

File: backend/typescript/services/implementations/schedulingService.ts (hunk 2 above)
Comment (linnall) [reply backend/typescript/services/implementations/schedulingService.ts]: Will refactor all logic for recurring schedule creation post-launch

**Pull request merged**